content/blog/LFX-Mentorship-Blog.md

```diff
@@ -0,0 +1,425 @@
+[gpub offramp](https://docs.tremor.rs/artefacts/offramps/#gpub) and [gsub onramp](https://docs.tremor.rs/artefacts/onramps/#gsub)
+
+## Walk-Through Guide
+
+To get started, you need a service account on GCP and you will need a GCP pem file for certificates authentication and a service token json file.
+
+The command used to get the service_account `json` file:
+```bash
+gcloud iam service-accounts keys create key-file.json -iam-account=<iam-account-name>@<project-id>.iam.gserviceaccount.com
+```
+or 
+```bash
+gcloud iam service-accounts keys create key-file.json -iam-account=<mail-id-of-service-account>
+```
+
+Go to GCP dashboard → IAM & Admin → Service Accounts and get the email-id mentioned `<mail-id-of-service-account> = email-id`
+
+### Google Cloud Storage
+
+The following files are required:
+
+**outbound.trickle**
+
+```trickle
+select event from in into out
+```
+
+**inbound.trickle**
+
+```trickle
+select event from in into out
+```
+**test.yaml**
+
+```yaml
+offramp:
+  - id: stdout
+    type: stdout
+    codec: json
+    config:
+  - id: gcs
+    type: gcs
+    codec: json
+    postprocessors:
+      - gzip    
+    preprocessors:
+      - gzip
+    linked: true
+    config:
+      pem: <path-to-pem-file>
+onramp:
+  - id: stdin
+    type: file
+    codec: json
+    config:
+      source: /dev/stdin   
+binding:
+  - id: example
+    links:
+      '/onramp/stdin/{instance}/out':
+        - '/pipeline/outbound/{instance}/in'
+      '/pipeline/outbound/{instance}/out':
+        - '/offramp/stdout/{instance}/in'
+        - '/offramp/gcs/{instance}/in'
+      '/offramp/gcs/{instance}/out':
+        - '/pipeline/inbound/{instance}/in'
+      '/pipeline/inbound/{instance}/out':
+        - '/offramp/stdout/{instance}/in'
+mapping:
+  "/binding/example/passthrough":
+    instance: "passthrough"
+```
+Supported preprocessors, that can be configured in yaml file can be found here: [preprocessors](https://docs.tremor.rs/artefacts/preprocessors/)
+
+Supported postprocessors and more about it: [postprocessors](https://docs.tremor.rs/artefacts/postprocessors/)
+
+Supported codecs, that can be configured in yam file can be found here: [codecs](https://docs.tremor.rs/artefacts/codecs/)
+
+- Set the env variable
+
+```bash
+export GOOGLE_APPLICATION_CREDENTIALS="<path-to-service-token-json-file>"
+```
+
+- Command used to run tremor:
+
+```bash
+ TREMOR_PATH=tremor-script/lib target/debug/tremor server run -f outbound.trickle inbound.trickle test.yaml | jq .
+```
+For a detailed guide on the operations that can be performed, refer the [docs](https://docs.tremor.rs/artefacts/offramps/#gcs).
+
+### Google Cloud Pub/sub
+
+Google Cloud Pub/Sub guarantees delivery of all messages, whether low throughput or high throughput, so there should be no concern about messages being lost.
+
+Pub/Sub guarantees at-least-once message delivery, which means that occasional duplicates are to be expected since we acknowledge the messages once they are received.
+
```

Comment:
Maybe add 1 or two sentences where you introduce that the following is a usage example and shows how to use you connector.

Reply:
Yes, actually this is all a part of the docs also - I did not include the entire walkthrough for GCS since the deployed version is updated but that's not the case for pub/sub. I can maybe just remove this for now and add- refer to the docs and add its link?

Reply:
Not the entire walkthrough but the part where I have added the commands we can use in pub/sub